### PR TITLE
feat: Components to show article in article search widget

### PR DESCRIPTION
**Pull request by nithindavid**
## Description

Article Loading component
image

image

Fixes # (issue)
https://linear.app/chatwoot/issue/CW-1454/component-to-render-full-article-within-search-container-view

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by sojan-official** [comment]
@nithindavid  Can we switch the styles to tailwind and get this merged

**Comment by pranavrajs** [comment]
@nithindavid Can you please cleanup the PR if you intend to get this merged?

**Comment by nithindavid** [comment]
@pranavrajs I'll get this ready.

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/ArticleView.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<template>
+  <div class="h-full w-full flex flex-col flex-1 overflow-hidden">
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/ArticleView.vue:2]: @nithindavid Is it possible to set a max height? Now it looks like this on bigger resolution screens. And can we reduce the gap between the editor and the article container (Only if the woot-editor is empty)?
image

File: app/javascript/dashboard/components/widgets/WootWriter/Editor.vue [hunk 2]
```diff
@@ -616,6 +605,44 @@ export default {
     onFocus() {
       this.$emit('focus');
     },
+    insertContentIntoEditor(content, defaultFrom = 0) {
+      const from = defaultFrom || this.editorView.state.selection.from || 0;
+      let node = new MessageMarkdownTransformer(messageSchema).parse(content);
+
+      this.insertNodeIntoEditor(node, from, undefined, content);
+    },
+    insertNodeIntoEditor(node, from = 0, to = 0, content = '') {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/WootWriter/Editor.vue:614]: @nithindavid Is it possible to move this method to the helper file? `app/javascript/dashboard/helper/messageEditorHelper.js`

**Review by iamsivin — COMMENTED** [review]
@nithindavid Please see the inline comments. And some suggestions, please feel free to add your thoughts.

1. Clicking outside the Article view modal doesn't close it. Can we add a click-away or ESC key shortcut to it?
2. Can we set a maximum height for the article view?
3. Can all articles be shown there on opening the article view modal? Now it's showing empty result text. But it shows if we search the article and clear the query.

image

4. Can we introduce arrow key shortcuts to insert the article in the editor? 
5. When we copy the article link. Can a toast or an icon change be shown when the copy to the clipboard succeeds?

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/ArticleView.vue (hunk 1 above)
Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/ArticleView.vue:2]: Sounds good. I'll set it 👍

**Comment by nithindavid** [comment]
Closing in favor of https://github.com/chatwoot/chatwoot/pull/8117/files

**Pull request closed without merging**